core/dbt/parser/manifest.py

```diff
@@ -1238,9 +1238,12 @@ def _process_semantic_information_for_metric(
             and not timestamp_col.time_grains
             and "default_time_grains" not in metric.config._extra.keys()
         ):
-            raise dbt.exceptions.CompilationException(
-                f"""The metric `{metric.name}` does not have time_grains configured. Please add a list of timegrains to the metric specification, the timestamp columns, or as a metric config."""
-            )
+            metric.time_grains = ["day", "week", "month", "year"]
```

Comment:
LGTM! If anything I think we can get rid of the compilation exception code instead of commenting it out.